lib/classes/class-bootstrap.php

```diff
@@ -1661,18 +1700,22 @@ public function run_install_process() {
        */
       public function run_upgrade_process() {
         // Creating database on new installation.
-        $this->create_db();
+        $this->create_sync_db();
+        ud_stateless_db()->create_db();
+
+        Migrator::instance()->migrate();
+        
         /**
          * Maybe Upgrade current Version
          */
         Upgrader::call($this->args['version']);
       }
 
       /**
-       * Create database on plugin activation.
+       * Create SYNC database on plugin activation.
        * @param boolean $force - whether to create db even if option exists. For debug purpose only.
        */
-      public function create_db($force = false) {
+      public function create_sync_db($force = false) {
```

Comment:
@balexey88 what do you think about moving it to `class-db.php`?

Reply:
`create_sync_db()` is called separately in [SyncNonMedia constructor](https://github.com/udx/wp-stateless/blob/aee8e2ccd15b1c83d353c84a0c29b8beefbc6628/lib/classes/class-sync-non-media.php#L21)

I'm not sure why it is made like this, so I'm just trying to keep sync untouched. And make new functionality not interfering the existing one. At least until we will have to change or refactor sync.

Reply:
We have to revise this functionality in this milestone because it is related to DB structure.

Original issue - https://github.com/udx/wp-stateless/issues/307

------

> create_sync_db() is called separately in [SyncNonMedia constructor](https://github.com/udx/wp-stateless/blob/aee8e2ccd15b1c83d353c84a0c29b8beefbc6628/lib/classes/class-sync-non-media.php#L21)

I do not see any reason to call it here. So I think it would be fine to call it once on `upgrade` action.

Reply:
Moved to a separate issue https://github.com/udx/www.stateless.ci/issues/92